### ISSUE TITLE
how to use Softmax when do segmentation

### ISSUE THREAD

**Issue by 994374821** [pytorch/pytorch#7946]
I want  implement FCN by caffe2 to do semantic segmentation,  but I do not konw how to implement Softmax as FCN in caffe version.
Could you help me, Thanks!
the code of FCN's softmax using caffe is as follows:
```
layer {
  name: "upscore"
  type: "Deconvolution"
  bottom: "score_fr"
  top: "upscore"
  param {
    lr_mult: 0
  }
  convolution_param {
    num_output: 21
    bias_term: false
    kernel_size: 64
    stride: 32
  }
}
layer {
  name: "score"
  type: "Crop"
  bottom: "upscore"
  bottom: "data"
  top: "score"
  crop_param {
    axis: 2
    offset: 19
  }
}
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "score"
  bottom: "label"
  top: "loss"
  loss_param {
    ignore_label: 255
    normalize: false
  }
}
```

**Comment by edubois** [comment]
Softmax is taking/producting a 2D tensor, you need a reshape on your logits.

**Comment by beichen2012** [comment]
"SpatialSoftmaxWithLoss" can compute the softmax and cross entropy along the "C" channel.
but: there is no crop op.....
